sardana_tango/ctrl/TangoAttrZeroDCtrl.py

```diff
@@ -68,32 +68,48 @@ def pre_read_one(self, axis):
-                                (dev, str(attributes)))
+        try:
+            for dev in list(self.devices_to_read.keys()):
+                attributes = self.devices_to_read[dev]
+                values = {}
+                try:
+                    dev_proxy = PoolUtil().get_device(self.GetName(), dev)
+                    # Set the list to prevent duplicated attr names
+                    # Tango raise exception on read_attributes if there are
+                    # duplicated attributes
+                    attrs = list(set(attributes))
+                    r_values = dev_proxy.read_attributes(attrs)
+                    values = dict(list(zip(attrs, r_values)))
+                except tango.DevFailed as e:
+                    # In case of DeviceServer error
+                    for attr in attributes:
+                        axis = self.axis_by_tango_attribute[dev + '/' + attr]
+                        self.devsExtraAttributes[axis][EVALUATED_VALUE] = e
+                    self._log.debug("Exception on read the attribute:%r"%e)
+                except Exception as e:
+                    self._log.error('Exception reading attributes:%s.%s' %
+                                    (dev, str(attributes)))
+            
             for attr in attributes:
-                axis = self.axis_by_tango_attribute[dev + '/' + attr]
-                formula = self.devsExtraAttributes[axis][FORMULA]
-                index = attributes.index(attr)
-                dev_attr_value = values[index]
-                if dev_attr_value.has_failed:
-                    VALUE = tango.DevFailed(*dev_attr_value.get_err_stack())
-                    self.devsExtraAttributes[axis][EVALUATED_VALUE] = VALUE
-                else:
-                    VALUE = float(dev_attr_value.value)
-                    # just in case 'VALUE' has been written in lowercase...
-                    value = VALUE
-                    self.devsExtraAttributes[axis][
-                        EVALUATED_VALUE] = eval(formula)
+                  axies = []
+                  for axis, dic in self.devsExtraAttributes.items():
+                      if dic[TANGO_ATTR] == dev+'/'+attr:
+                           axies.append(axis)
+                  for axis in axies:
+                    if len(values) > 0:
+                        dev_attr_value = values[attr]
+                        if dev_attr_value.has_failed:
+                            # In case of Attribute error
+                            VALUE = tango.DevFailed(*dev_attr_value.get_err_stack())
+                            self.devsExtraAttributes[axis][EVALUATED_VALUE] = VALUE
+                        else:
+                            formula = self.devsExtraAttributes[axis][FORMULA]
+                            VALUE = float(dev_attr_value.value)
+                            value = VALUE # just in case 'VALUE' has been written in lowercase...
+                            v = eval(formula)
```

Comment:
You can change to:
value = float(dev_attr_value.value)
v = eval(formula, {'value': value, 'VALUE': value, 'Value':value}